domain/network/state/types_test.go

```diff
@@ -188,3 +188,11 @@ func getNetAddressTypes() nameToIDTable {
 func ptr[T any](v T) *T {
 	return &v
 }
+
+func ptrZeroAsNil[T comparable](v T) *T {
```

Comment:
suggestion: It's helpful if these little helpers have consistent naming across domains, e.g `ptr` is consistent across domains.

`nilZeroPtr` has already been used in the machine domain.